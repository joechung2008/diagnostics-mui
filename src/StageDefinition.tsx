import {
  Table,
  TableBody,
  TableCell,
  TableHead,
  TableRow,
  Typography,
} from "@mui/material";

const StageDefinition: React.FC<StageDefinitionProps> = ({
  stageDefinition,
}) => {
  const items = Object.entries(stageDefinition).reduce<
    KeyValuePair<string[]>[]
  >((previous, [key, value]) => [...previous, { key, value }], []);

  return (
    <div>
      <Typography variant="h3" fontWeight="semibold">
        Stage Definitions
      </Typography>
      <Table aria-label="Stage Definitions">
        <TableHead>
          <TableRow>
            <TableCell>Key</TableCell>
            <TableCell>Value</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {items.map((item) => (
            <TableRow key={item.key}>
              <TableCell>{item.key}</TableCell>
              <TableCell>{item.value.join(", ")}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </div>
  );
};

export default StageDefinition;
