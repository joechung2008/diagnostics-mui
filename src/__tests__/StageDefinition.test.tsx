import { render } from "@testing-library/react";
import { describe, expect, it } from "vitest";
import StageDefinition from "../StageDefinition";

describe("StageDefinition", () => {
  it("matches snapshot", () => {
    const { container } = render(
      <StageDefinition stageDefinition={{ stage1: ["step1", "step2"] }} />
    );
    expect(container).toMatchSnapshot();
  });
});
