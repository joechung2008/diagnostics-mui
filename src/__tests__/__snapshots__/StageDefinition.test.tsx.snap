// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`StageDefinition > matches snapshot 1`] = `
<div>
  <div>
    <h3
      class="MuiTypography-root MuiTypography-h3 css-1lw8d58-MuiTypography-root"
    >
      Stage Definitions
    </h3>
    <table
      aria-label="Stage Definitions"
      class="MuiTable-root css-1xwxv7r-MuiTable-root"
    >
      <thead
        class="MuiTableHead-root css-1a7iywq-MuiTableHead-root"
      >
        <tr
          class="MuiTableRow-root MuiTableRow-head css-1xtozoh-MuiTableRow-root"
        >
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
            scope="col"
          >
            Key
          </th>
          <th
            class="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1orzuox-MuiTableCell-root"
            scope="col"
          >
            Value
          </th>
        </tr>
      </thead>
      <tbody
        class="MuiTableBody-root css-gmh7jj-MuiTableBody-root"
      >
        <tr
          class="MuiTableRow-root css-1xtozoh-MuiTableRow-root"
        >
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
          >
            stage1
          </td>
          <td
            class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1dc80h3-MuiTableCell-root"
          >
            step1, step2
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
